weave/tests/test_exec.py

```diff
@@ -0,0 +1,16 @@
+import textwrap
+
+
+def test_publish_works_for_code_with_no_source_file(client):
+    code = textwrap.dedent(
+        """
+        import weave
+        
+        @weave.op()
+        def add(a: int, b: int) -> int:
+            return a + b
+
+        ref = weave.publish(add)
+        """
+    )
+    exec(code)
```

Comment:
is there a way to get the "saved op code" back?

Reply:
You can do this: https://github.com/wandb/weave-analysis/blob/7d5ea67862f610ea220c547430bc25baad2a08c3/robot.py#L120.

Not a very nice API yet...